chats/views.py

```diff
@@ -39,55 +43,79 @@ def index(request):
-            return redirect('register')
+            #messages.error(request, "Username must be under 20 characters!")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Username must be under 20 characters!"})
         
         if pass1 != pass2:
-            messages.error(request, "Passwords didn't matched!")
-            return redirect('register')
+            #messages.error(request, "Passwords didn't matched!")
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Passwords didn't matched!"})
+
+        passReg = "^(?=.*[a-z])(?=.*[A-Z])(?=.*\d)(?=.*[@$!%*#?&])[A-Za-z\d@$!#%*?&]{8,}$"
+        pat = re.compile(passReg)
+        if not re.search(pat, pass1):
+            return JsonResponse({"status" : "unsuccessfull",
+                                "error": "Your password must contain at least one small letter, one capital letter, one number, one special symbol and must be at least 8 symbols long."})
 
         myusr = User.objects.create_user(username, email, pass1)
         myusr.first_name = fname
         myusr.last_name = lname
         myusr.is_active = True
         myusr.save()
-        messages.success(request, "Your account has been created succesfully!")
-        return redirect("login")
-
-    return render(request, "register.html") 
+        #messages.success(request, "Your account has been created succesfully!")
+        return JsonResponse({"status" : "successfull"})
 
+@csrf_exempt
 def login(request):
     if request.method == "POST":
-        username = request.POST['usrname']
-        password = request.POST['pass']
-        user = authenticate(request, username=username, password=password)
+        usrname = json.loads(request.body)["username"]
+        password = json.loads(request.body)['password']
+        print(password)
+        user = authenticate(request, username=usrname, password=password)
+        print(user)
         if user is not None:
             auth_login(request, user)
             print("logged")
-            return redirect("index")
+            current_user = request.user
+            return JsonResponse({'status':'successfull',
+                                    "id" : current_user.id,
+                                    "password": current_user.password,
+                                    "username": current_user.username,
+                                    "email": current_user.email,
+                                    "first_name": current_user.first_name,
+                                    "last_name": current_user.last_name,
+                                    "is_active": current_user.is_active})
         else:
             print("not logged")
-            return render(request, "login.html")
-    else:
-       return render(request, "login.html") 
+            return JsonResponse({"status" : "unsuccessfull"})
 
 def log_out(request):
     logout(request)
```

Comment:
Please comment the way your body looks. Whaen I try the /login request with:
{
    "username": "Marto",
    "password": "12345678"
}
It returns:
{
    "status": "unsuccessfull"
}
So I guess the reques might have to be different.

Reply:
"Marto" with password "12345678" is existing user. The server loggs correct received username and password.

Reply:
good we fixed it